akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorSystemAdapter.scala

```diff
@@ -74,7 +75,7 @@ import akka.event.LoggingFilterWithMarker
   }
   override def dynamicAccess: untyped.DynamicAccess = untypedSystem.dynamicAccess
   implicit override def executionContext: scala.concurrent.ExecutionContextExecutor = untypedSystem.dispatcher
-  override val log: Logger = new LoggerAdapterImpl(
+  override lazy val log: Logger = new LoggerAdapterImpl(
     untypedSystem.eventStream,
     getClass,
     name,
```

Comment:
> There is still the case where an extension is eagerly constructed and tries to use for example the ActorSystemAdapter in its constructor, but I think that problem is also present in untyped so maybe that is ok.

Could you elaborate on this. What combination is that? An untyped extension that is using system.toTyped? A typed extension is always ok?

Reply:
The case I can think of is when the extension is eager-loaded through config and uses the ActorAdapter from its constructor.

Reply:
Made eager and covered with a test case